File: 1-TwoSum/1-TwoSum.py
# Last updated: 8/4/2026, 9:57:00 AM
1class Solution(object):
2    def mySqrt(self, x):
3        if x == 0:
4            return 0
5        left, right = 1, x
6        while left <= right:
7            mid = (left + right) // 2
8            if mid * mid == x:
9                return mid
10            elif mid * mid < x:
11                left = mid + 1
12            else:
13                right = mid - 1
14        return right

File: 1-TwoSum/1-TwoSum.java
// Last updated: 8/4/2026, 9:56:04 AM
1class Solution {
2    public int[] plusOne(int[] digits) {
3     for (int i = digits.length - 1; i >= 0; i--) {
4	if (digits[i] < 9) {
5		digits[i]++;
6		return digits;
7	}
8	digits[i] = 0;
9}
10
11digits = new int[digits.length + 1];
12digits[0] = 1;
13return digits;   
14    }
15}